src/java.base/share/classes/java/util/Properties.java

```diff
@@ -833,6 +876,15 @@ public void save(OutputStream out, String comments)  {
      * After the entries have been written, the output stream is flushed.
      * The output stream remains open after this method returns.
      *
+     * @implNote When writing the date comment, this method checks whether the
+     * {@code SOURCE_DATE_EPOCH} environment variable is set. If it is set, then instead
+     * of writing the current date and time, the date and time represented by the
+     * value of {@code SOURCE_DATE_EPOCH} will be written, using the
+     * {@code EEE MMM dd HH:mm:ss zzz yyyy} {@link DateTimeFormatter date format} with a
+     * {@link Locale#ROOT root locale} and {@link ZoneOffset#UTC UTC zone offset}.
+     * If the value set for {@code SOURCE_DATE_EPOCH} cannot be parsed to a {@code long},
+     * then the current date and time will be written.
+     *
```

Comment:
Update to refer to the property and add the {@systemProperty ...} javadoc tag so the property is listed in the system property reference.

Reply:
"different set implementation" as part of the spec may challenge the compatibility test developers to prove or disprove that statement.
The type of an instance is frequently understood to be the "implementation".
The visible type returned from Properties.entrySet() is SynchronizedSet.
Anyone can create one of those.
That statement might mislead a subclass into thinking they can not/must not return a SynchronizedSet if they want the built-in sorting.
I'm thinking that wording it in other term of the subclass might be better:
"...sort order of the keys in the entrySet() unless entrySet() is overridden by a subclass to return a different value than 'super.entrySet'().   "
The existing implementation is fine.

Reply:
I agree that this is a better formulation than what I suggested :-)

Reply:
Done. I've updated the PR to use Roger's suggested text and yes it's much more precise than what we had so far.